enso/experiment/__init__.py

```diff
@@ -162,8 +167,32 @@ def _run_sub_experiment(
 
     def _run_experiment(self, dataset_name, current_setting, experiments):
         """Responsible for running all experiments specified in config."""
-        results = pd.DataFrame(columns=self.columns)
         dataset = self._load_dataset(dataset_name, current_setting.get("Featurizer"))
+        if EXPERIMENT_PARAMS:
+            exp_params = deepcopy(EXPERIMENT_PARAMS)
+            # add any experiments that were not included
+            for experiment_cls in experiments:
+                if experiment_cls.__name__ not in exp_params:
+                    exp_params[experiment_cls.__name__] = {}
+            # add the 'All' configuration to all the experiments
+            if 'All' in EXPERIMENT_PARAMS.keys():
+                for k, v in exp_params.items():
+                    if k != 'All':
+                        v.update(EXPERIMENT_PARAMS['All'])
+            del exp_params['All']
+            hparams_by_experiment = {
+                exp_name: list(ParameterGrid(hparams)) for exp_name, hparams in exp_params.items()
+            }
+            # make sure all experiments share the same experiment params
+            param_keys = list(list(exp_params.values())[0].keys())
```

Comment:
Is this a necessary constraint?  Couldn't we sweep over different params for each model?

Reply:
To make this work with the visualization would could maybe do something like store hparam settings as a formatted string in a "hyperparameter_setting" column or something.

Reply:
this just ensures that all the models have at least a value specified for each of the parameter keys, not that the values are always the same per model. I changed the comment to reflect that